sockeye/utils.py

```diff
@@ -615,3 +615,52 @@ def all_gather_object(obj: T) -> List[T]:
     obj_list = [None] * torch.distributed.get_world_size()  # type: List[T]
     torch.distributed.all_gather_object(obj_list, obj)
     return obj_list
+
+
+def count_seq_len(sample: str, count_type: str = 'char', replace_tokens: list = []) -> int:
+    """Count sequence length (char, tokens) with optional replace tokens.
+    If count_type not char, count tokens."""
+    if len(replace_tokens) >= 1:
+        for r in replace_tokens:
+            sample = sample.replace(r, '')
+    if count_type != 'char':
```

Comment:
this is confusing to me. If the type is not `char`, return the number of characters?

Reply:
thanks for catching this, updated to handle both character and token counts.